File: client-web/src/app/core/models/subscription.models.ts
export enum BillingCycle {
  Monthly = 0,
  Quarterly = 1,
  Yearly = 2,
  HalfYearly = 3
}

export enum SubscriptionStatus {
  Active = 0,
  Paused = 1,
  Cancelled = 2,
  Expired = 3
}

export interface SubscriptionPlanFeature {
  id: string;
  text: string;
  displayOrder: number;
}

export interface SubscriptionPlan {
  id: string;
  name: string;
  description?: string;
  billingCycle: BillingCycle;
  garmentsPerCycle: number;
  price: number;
  displayOrder: number;
  isActive: boolean;
  features: SubscriptionPlanFeature[];
}

export interface CreateSubscriptionPlanRequest {
  name: string;
  description?: string;
  billingCycle: BillingCycle;
  garmentsPerCycle: number;
  price: number;
  displayOrder: number;
  features: string[];
}

export interface UpdateSubscriptionPlanRequest extends CreateSubscriptionPlanRequest {
  isActive: boolean;
}

export interface CustomerSubscriptionListItem {
  id: string;
  customerId: string;
  customerName: string;
  subscriptionPlanId: string;
  planName: string;
  recurringValue: number;
  startDate: string;
  endDate: string;
  nextBillingDate?: string;
  status: SubscriptionStatus;
  notes?: string;
}

export type CustomerSubscriptionDetail = CustomerSubscriptionListItem;

export interface AssignCustomerSubscriptionRequest {
  customerId: string;
  subscriptionPlanId: string;
  recurringValue: number;
  startDate: string;
  notes?: string;
}

export interface UpdateCustomerSubscriptionRequest {
  recurringValue: number;
  status: SubscriptionStatus;
  nextBillingDate?: string | null;
  notes?: string;
}

File: LaundryMgmt.Mobile/Models/SubscriptionModels.cs
namespace LaundryMgmt.Mobile.Models;

public enum BillingCycle
{
    Monthly = 0,
    Quarterly = 1,
    Yearly = 2,
    HalfYearly = 3
}

public record SubscriptionPlanFeatureDto(Guid Id, string Text, int DisplayOrder);

public record SubscriptionPlanDto(
    Guid Id, string Name, string? Description, BillingCycle BillingCycle, int GarmentsPerCycle,
    decimal Price, int DisplayOrder, bool IsActive, List<SubscriptionPlanFeatureDto> Features);

public record CreateSubscriptionPlanRequest(
    string Name, string? Description, BillingCycle BillingCycle, int GarmentsPerCycle,
    decimal Price, int DisplayOrder, List<string> Features);

public record UpdateSubscriptionPlanRequest(
    string Name, string? Description, BillingCycle BillingCycle, int GarmentsPerCycle,
    decimal Price, int DisplayOrder, bool IsActive, List<string> Features);

public enum SubscriptionStatus
{
    Active = 0,
    Paused = 1,
    Cancelled = 2,
    Expired = 3
}

public record CustomerSubscriptionListItemDto(
    Guid Id, Guid CustomerId, string CustomerName, Guid SubscriptionPlanId, string PlanName,
    decimal RecurringValue, DateOnly StartDate, DateOnly EndDate, DateOnly? NextBillingDate,
    SubscriptionStatus Status, string? Notes);
